cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -149,10 +169,71 @@ def step(self):
             raise DestinationNotFound("Nodes aren't present on details page of this provider")
 
 
+class ProviderRegisterNodesView(View):
+    """
+     represents Register Nodes view
+    """
+    file = FileInput(locator='//input[@id="nodes_json_file"]')
+    register = Button(value='Register')
+    cancel = Button(value='Cancel')
+
+    @property
+    def is_displayed(self):
+        return False
+
+
+class ProviderScaleDownView(View):
+    """
+     represents Scale down view
+    """
+    table = Table(locator='//div[contains(@class, "form-horizontal")]//table')
```

Comment:
**OPTIONAL**
I would rely on some more unique path. Try to find parent div with some id.